core/src/main/java/com/linecorp/armeria/server/HttpResponseSubscriber.java

```diff
@@ -414,15 +408,8 @@ public boolean canSchedule() {
 
             @Override
             public void run() {
-                if (state != State.DONE) {
```

Comment:
Could you explain why we don't need this condition please?

Reply:
Because `canSchedule()` checks the condition already.
`TimeoutScheduler.invokeTimeoutTask()` checks the returned value of `canSchedule()` before executing a timeout task.

Reply:
Then maybe better adding some assertion and comments?